WordPress/WordPressShareExtension/SharePostTypePickerViewController.swift

```diff
@@ -0,0 +1,195 @@
+        configurePostTypeCell(cell, indexPath: indexPath)
+        return cell
+    }
+
+    override func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        return Constants.defaultRowHeight
+    }
+
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return nil
+    }
+
+    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        return nil
+    }
+
+    override func tableView(_ tableView: UITableView, willDisplayFooterView view: UIView, forSection section: Int) {
+        WPStyleGuide.configureTableViewSectionFooter(view)
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.flashRowAtIndexPath(indexPath,
+                                      scrollPosition: .none,
+                                      flashLength: Constants.flashAnimationLength,
+                                      completion: nil)
+        selectedPostTypeTableRowAt(indexPath)
+    }
+}
+
+// MARK: - Post Type UITableView Helpers
+
+fileprivate extension SharePostTypePickerViewController {
+    func configurePostTypeCell(_ cell: UITableViewCell, indexPath: IndexPath) {
+        let postType = postTypeForRowAtIndexPath(indexPath)
+        cell.textLabel?.text = postType.title
+        cell.detailTextLabel?.isEnabled = false
+        cell.detailTextLabel?.text = nil
+        if selectedPostType == postType {
+            cell.accessoryType = .checkmark
+        } else {
+            cell.accessoryType = .none
+        }
+        WPStyleGuide.Share.configurePostTypeCell(cell)
+    }
+
+    var rowCountForPostTypes: Int {
+        return PostType.allCases.count
+    }
+
+    func selectedPostTypeTableRowAt(_ indexPath: IndexPath) {
+        guard let cell = tableView.cellForRow(at: indexPath) else {
+            return
+        }
+
+        deselectedPostTypeTableRowAt(indexPathForPostType(selectedPostType))
+
+        let postType = postTypeForRowAtIndexPath(indexPath)
+        selectedPostType = postType
+        cell.accessoryType = .checkmark
+    }
+
+    func deselectedPostTypeTableRowAt(_ indexPath: IndexPath) {
+        guard let cell = tableView.cellForRow(at: indexPath) else {
+            return
+        }
+        cell.accessoryType = .none
+    }
+
+    func postTypeForRowAtIndexPath(_ indexPath: IndexPath) -> PostType {
+        return PostType.allCases[indexPath.row]
+    }
+
+    func indexPathForPostType(_ postType: PostType) -> IndexPath {
+        let postTypeRow = PostType.allCases.firstIndex(of: postType)!
+        return IndexPath(row: postTypeRow, section: 0)
+    }
+}
+
+// MARK: - Actions
+
+extension SharePostTypePickerViewController {
+    @objc func cancelWasPressed() {
+        _ = navigationController?.popViewController(animated: true)
```

Comment:
There's some space to improve this line of code.

Reply:
So with this Picker, I've tried to keep it as close to the other pickers as possible. This line appears in those, too. It does seem like, for this new picker, I should probably not do it like this (others must have been written before `popViewController` was annotated with `@discardableResult`), but what should my scope be things like this? 

I can see multiple other instances of this, including outside of the Share Extension - should I correct it only in my new file, in all files in the Extension, or across the app?

Reply:
The scope, as I see it, is to make any added code clean and tight.  I'd definitely avoid making changes throughout the App as it'd be far beyond the scope of this PR.

I agree though that this probably was added like this back when the result wasn't `@discardableResult`, but that's just a guess on my side.

But even then, when copying code from elsewhere my take is that we're signing it, as both the editor and reviewer.

That's not going to save us from having legacy code that's outdated, but at least we can do something for the new code!

Thanks for the explanation!

Reply:
Agreed on all points, I'll tidy up what I've added 👍